src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -219,94 +216,28 @@ MyCodeAction CreateNewCodeAction(string actionName, bool allOccurrences, Introdu
             IMethodSymbol methodSymbol, SyntaxNode containingMethod, bool allOccurrences, IntroduceParameterCodeActionKind selectedCodeAction,
             CancellationToken cancellationToken)
         {
-            var parameterName = await GetNewParameterNameAsync(originalDocument, expression, cancellationToken).ConfigureAwait(false);
-
             var methodCallSites = await FindCallSitesAsync(originalDocument, methodSymbol, cancellationToken).ConfigureAwait(false);
 
             var modifiedSolution = originalDocument.Project.Solution;
-            var mappingDictionary = await MapExpressionToParametersAsync(originalDocument, expression, cancellationToken).ConfigureAwait(false);
+            var generator = SyntaxGenerator.GetGenerator(originalDocument);
             var syntaxFacts = originalDocument.GetRequiredLanguageService<ISyntaxFactsService>();
+            var semanticFacts = originalDocument.GetRequiredLanguageService<ISemanticFactsService>();
```

Comment:
these values are computed just to pass into the rewriter.  but you pass in teh value these are derived from.  so this logic can be pushed into that constructor.

Reply:
moved